pkg/dogstatsd/server.go

```diff
@@ -69,7 +69,7 @@ func NewServer(metricOut chan<- *metrics.MetricSample, eventOut chan<- metrics.E
 		stats = s
 	}
 
-	packetChannel := make(chan *listeners.Packet, 100)
+	packetChannel := make(chan listeners.Packets, 100)
```

Comment:
With the `PacketBuffer` we're now increasing the amount of traffic in-flight (potentially) by the size of said buffers, so, potentially up to 100*4096 (currently non-configurable, but I guess we might change that). How will that play with the PacketPool below where we use `dogstatsd_buffer_size`? We should probably put some thought into how we should put this all together so all these settings work well together.

Reply:
You're right, we need to be careful or the memory we allocate might skyrocket if the pipeline can't keep up. I need to think about it

Reply:
For `dogstatsd_buffer_size`, it does not control the size of the `Packet` object poll if that's what you were thinking but the default inner buffer size of the packets (which I don't think we should worry about here ?) : 
https://github.com/DataDog/datadog-agent/blob/f352b413deb30ac031e48036bcdbe2b329c3bcb6/pkg/dogstatsd/listeners/packet_pool.go#L30

The number of packet objects is automagically managed by `sync.Pool`